textattack/transformations/word_swap_masked_lm.py

```diff
@@ -0,0 +1,209 @@
+import itertools
+
+import numpy as np
+import torch
+from transformers import AutoModelWithLMHead, AutoTokenizer
+
+from textattack.shared import utils
+from textattack.transformations.word_swap import WordSwap
+
+
+class WordSwapMaskedLM(WordSwap):
+    """
+    Generate potential replacements for a word using BERT-Masked LM.
+
+    Based off of two papers
+        - "BAE: BERT-based Adversarial Examples for Text Classification" (Garg et al., 2020) https://arxiv.org/abs/2004.01970
+        - "BERT-ATTACK: Adversarial Attack Against BERT Using BERT" (Li et al, 2020) https://arxiv.org/abs/2004.09984
+
+    Choose which method to use by specifying "bae" or "bert-attack" for `method` argument
+    """
+
+    def __init__(
+        self,
+        method="bae",
+        masked_language_model="bert-base-uncased",
+        max_length=256,
+        max_candidates=50,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.method = method
+        self.masked_lm_name = masked_language_model
+        self.max_length = max_length
+        self.max_candidates = max_candidates
+
+        self._lm_tokenizer = AutoTokenizer.from_pretrained(
+            masked_language_model, use_fast=True
+        )
+        self._language_model = AutoModelWithLMHead.from_pretrained(
+            masked_language_model
+        )
+        self._language_model.to(utils.device)
+        self._segment_tensor = (
+            torch.zeros(self.max_length, dtype=torch.long).unsqueeze(0).to(utils.device)
+        )
+        self._language_model.eval()
+
+    def _bae_replacement_words(self, current_text, index):
+        # TODO: is it necessary to create a new AttackedText to recover case and punctuation?
+        masked_attacked_text = current_text.replace_word_at_index(
+            index, self._lm_tokenizer.mask_token
+        )
+        ids = self._lm_tokenizer.encode(
+            masked_attacked_text.text,
+            max_length=self.max_length,
+            pad_to_max_length=True,
+        )
+
+        try:
+            masked_index = ids.index(self._lm_tokenizer.mask_token_id)
```

Comment:
when can this happen?

Reply:
Since default `max_length=256`, tokenizer would cut off all tokens include mask token that are past 256. If the word we want to change is past 256, then we run into this issue.